src/lib/index.ts

```diff
@@ -0,0 +1,272 @@
+          sub: this.sub,
+          key: this.key,
+          keyFile: this.keyFile,
+          scope: this.scope
+        });
+        callback();
+      });
+    } else {
+      callback(new Error('No token to revoke.'));
+    }
+  }
+
+  /**
+   * Configure the GoogleToken for re-use.
+   * @param  {object} options Configuration object.
+   */
+  private configure(options?: TokenOptions) {
+    options = options || {};
+    this.keyFile = options.keyFile;
+    this.key = options.key;
+    this.token = this.expiresAt = this.rawToken = null;
+    this.iss = options.email || options.iss;
+
+    if (options.sub) {
+      this.sub = options.sub;
+    }
+
+    if (typeof options.scope === 'object') {
+      this.scope = options.scope.join(' ');
+    } else {
+      this.scope = options.scope;
+    }
+  }
+
+  /**
+   * Request the token from Google.
+   *
+   * @param  {Function} callback The callback function.
+   */
+  private requestToken(
+      callback: (err: Error|null, token: string|null) => void) {
+    const iat = Math.floor(new Date().getTime() / 1000);
+    const payload = <Payload>{
+      iss: this.iss,
+      scope: this.scope,
+      aud: GOOGLE_TOKEN_URL,
+      exp: iat + 3600,  // 3600 seconds = 1 hour
+      iat: iat,
+    };
+
+    if (this.sub) {
+      payload.sub = this.sub;
+    }
+
+    const toSign = {
+      header: {alg: 'RS256', typ: 'JWT'},
+      payload: payload,
+      secret: this.key
+    };
+
+    return this.signJWT(toSign, (err, signedJWT) => {
+      if (err) {
+        callback(err, null);
+        return;
+      }
+
+      return request(
+          {
+            method: 'post',
+            url: GOOGLE_TOKEN_URL,
+            form: {
+              grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+              assertion: signedJWT
+            }
+          },
+          (err2: Error, res: request.RequestResponse, body: any) => {
+            try {
+              body = JSON.parse(body);
+            } catch (e) {
+              body = {};
+            }
+
+            err2 = err2 ||
+                body.error &&
+                    new Error(
+                        body.error +
+                        (body.error_description ?
+                             ': ' + body.error_description :
+                             ''));
+
+            if (err2) {
+              this.token = null;
+              this.tokenExpires = null;
+              callback(err2, null);
+              return;
+            }
+
+            this.rawToken = body;
+            this.token = body.access_token;
+            this.expiresAt = (iat + body.expires_in) * 1000;
+            return callback(null, this.token);
+          });
+    });
+  }
+
+  /**
+   * Sign the JWT object, returning any errors in the callback.
+   *
+   * @param opts     The configuration object.
+   * @param callback The callback function.
+   */
+  private signJWT(
+      opts: any, callback: (err: Error|null, result?: string|null) => void) {
```

Comment:
nit for follow-on: This looks async, but is actually a synchronous function.

Reply:
will handle in #21